src/main/java/com/crafting/Knowledge.java

```diff
@@ -0,0 +1,45 @@
+package com.crafting;
+
+import com.minecolonies.util.InventoryUtils;
+import net.minecraft.inventory.InventoryCrafting;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.crafting.IRecipe;
+import net.minecraft.world.World;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.List;
+
+
+/**
+ * One bit of knowledge from the crafting system.
+ */
+public class Knowledge
+{
+    @NotNull
+    private final IRecipe         recipe;
+    @NotNull
+    private final List<ItemStack> requirements;
+    @NotNull
+    private final ItemStack       output;
+
+    public static Knowledge tryFromRecipeAndGrid(@NotNull final InventoryCrafting craftingGrid, @NotNull final IRecipe recipe, @NotNull World world)
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Document this public method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AUndocumentedApi)